src/ansys/tools/path/path.py

```diff
@@ -353,102 +557,219 @@ def save_ansys_path(exe_loc=None, allow_prompt=True):
+        Version of Ansys MAPDL to search for. For example ``version=22.2``.
         If ``None``, use latest.
 
     """
-    exe_loc = None
-    if not version and os.path.isfile(CONFIG_FILE):
-        with open(CONFIG_FILE) as f:
-            exe_loc = f.read()
-        # verify
-        if not os.path.isfile(exe_loc) and allow_input:
-            exe_loc = save_ansys_path()
-    elif not version and allow_input:  # create configuration file
-        exe_loc = save_ansys_path()
+    return _get_application_path("mapdl", allow_input, version)
 
-    if exe_loc is None:
-        exe_loc = find_ansys(version=version)[0]
-        if not exe_loc:
-            exe_loc = None
 
-    return exe_loc
+def get_ansys_path(allow_input: bool = True, version=None) -> str:
+    """Deprecated, use `get_mapdl_path` instead"""
+    return _get_application_path("mapdl", allow_input, version)
+
+
+def get_mechanical_path(allow_input=True, version=None) -> str:
+    """Acquires Ansys Mechanical Path from a cached file or user input
+
+    Parameters
+    ----------
+    allow_input : bool, optional
+        Allow user input to find Ansys Mechanical path.  The default is ``True``.
+
+    version : float, optional
+        Version of Ansys Mechanical to search for. For example ``version=22.2``.
+        If ``None``, use latest.
+
+    """
+    return _get_application_path("mechanical", allow_input, version)
+
+
+def _mechanical_version_from_path(path):
+    """Extract the Ansys Mechanical version from a path.
+
+    Generally, the version of Mechanical is contained in the path:
+
+    - On Windows, for example: ``C:/Program Files/ANSYS Inc/v231/aisol/bin/winx64/AnsysWBU.exe``
+    - On Linux, for example: ``/usr/ansys_inc/v231/aisol/.workbench``
+
+    Parameters
+    ----------
+    path : str
+        Path to the Mechanical executable file.
+
+    Returns
+    -------
+    int
+        Integer version number (for example, 231).
+
+    """
+    # expect v<ver>/ansys
+    # replace \\ with / to account for possible windows path
+    matches = re.findall(r"v(\d\d\d)", path.replace("\\", "/"), re.IGNORECASE)
+    if not matches:
+        raise RuntimeError(f"Unable to extract Mechanical version from {path}.")
+    return int(matches[-1])
+
+
+def version_from_path(product, path):
+    """Extract the product version from a path.
+
+    Parameters
+    ----------
+    path : str
+        Path to the executable file.
+
+    Returns
+    -------
+    int
+        Integer version number (for example, 231).
+
+    """
+    if product == "mechanical":
+        return _mechanical_version_from_path(path)
+    raise Exception("Unexpected product")
+
+
+def check_valid_mechanical():
+    """Check if a valid version of Mechanical is installed and preconfigured."""
+    mechanical_bin = _get_application_path("mechanical", False)
+    if mechanical_bin is not None:
+        version = version_from_path(mechanical_bin)
+        return not (version < 231 and os.name != "posix")
```

Comment:
```suggestion
        return not (version < MINIMUM_MECHANICAL_VERSION and os.name != "posix")
```

Probably we should have something like this defined at the beginning of the file. Although I dont expect it to change.

Reply:
@germa89 actually I would rather not do this check here, since ansys-tools-path is more general than pymechanical.  someone may use it to run mechanical from command line rather than use the pyansys client